src/AspNetCoreModuleV2/AspNetCore/applicationinfo.cpp

```diff
@@ -66,48 +69,70 @@ APPLICATION_INFO::CreateHandler(
 HRESULT
 APPLICATION_INFO::CreateApplication(const IHttpApplication& pHttpApplication)
 {
-    HRESULT hr = S_OK;
-
     if (AppOfflineApplication::ShouldBeStarted(pHttpApplication))
     {
         LOG_INFO(L"Detected app_offline file, creating polling application");
-        #pragma warning( push )
-        #pragma warning ( disable : 26409 ) // Disable "Avoid using new", using custom deleter here
-        m_pApplication.reset(new AppOfflineApplication(pHttpApplication));
-        #pragma warning( pop )
+        m_pApplication.reset(std::make_unique<AppOfflineApplication>(pHttpApplication).release());
+        return S_OK;
     }
     else
     {
-        FINISHED_IF_FAILED(m_handlerResolver.GetApplicationFactory(pHttpApplication, m_pApplicationFactory));
+        try
+        {
+            const WebConfigConfigurationSource configurationSource(m_pServer.GetAdminManager(), pHttpApplication);
+            ShimOptions options(configurationSource);
 
-        LOG_INFO(L"Creating handler application");
-        IAPPLICATION * newApplication;
-        FINISHED_IF_FAILED(m_pApplicationFactory->Execute(
-            &m_pServer,
-            &pHttpApplication,
-            &newApplication));
+            const auto hr = TryCreateApplication(pHttpApplication, options);
 
-        m_pApplication.reset(newApplication);
+            if (FAILED_LOG(hr))
+            {
+                // Log the failure and update application info to not try again
+                EventLog::Error(
+                    ASPNETCORE_EVENT_ADD_APPLICATION_ERROR,
+                    ASPNETCORE_EVENT_ADD_APPLICATION_ERROR_MSG,
+                    pHttpApplication.GetApplicationId(),
+                    hr);
+
+                m_pApplication.reset(std::make_unique<ServerErrorApplication>(
+                    pHttpApplication,
+                    hr,
+                    options.QueryDisableStartupPage(),
+                    options.QueryHostingModel() == APP_HOSTING_MODEL::HOSTING_IN_PROCESS ? IDR_HTML1 : IDR_HTML2)
```

Comment:
Better resource names. and maybe ServerErrorApplication ctor overload that takes optional options reference.

Reply:
Made another constructor that didn't take either disableStartupPage or Hosting model.